temba/msgs/models.py

```diff
@@ -1045,12 +1074,17 @@ def mark_sent(cls, r, msg, status, external_id=None):
             Msg.objects.filter(id=msg.id).update(status=status, sent_on=msg.sent_on)
 
     def as_json(self):
-        return dict(direction=self.direction,
-                    text=self.text,
-                    id=self.id,
-                    attachments=self.attachments,
-                    created_on=self.created_on.strftime('%x %X'),
-                    model="msg")
+        msg_json = dict(direction=self.direction,
+                        text=self.text,
+                        id=self.id,
+                        attachments=self.attachments,
+                        created_on=self.created_on.strftime('%x %X'),
+                        model="msg")
+
+        if self.metadata:
+            msg_json['metadata'] = self.metadata
```

Comment:
I think we load this and include it as a nested dict

Reply:
Yeah that's probably simpler - so if it isn't set in a message, it should go into the payload as `{}`